src/main/java/org/candlepin/model/AnonymousCloudConsumerCurator.java

```diff
@@ -99,6 +99,26 @@ public AnonymousCloudConsumer getByCloudInstanceId(String instanceId) {
         return (AnonymousCloudConsumer) criteria.uniqueResult();
     }
 
+    /**
+     * Retrieves an anonymous cloud consumer using the provided cloud instance ID.
+     *
+     * @param accountId
+     *     the ID of the cloud instance that used to retrieve an anonymous cloud consumer
+     *
+     * @return the anonymous consumer based on the cloud instance ID, or null if no matching anonymous
+     * cloud consumer is found
+     */
+    public List<AnonymousCloudConsumer> getByCloudAccountId(String accountId) {
+        if (accountId == null || accountId.isBlank()) {
+            return new ArrayList<>();
+        }
+
+        Criteria criteria = this.createSecureCriteria()
```

Comment:
Please do not use the 'secure' criteria, they're deprecated and we're trying to get rid of them. Same for Hibernate Criteria API (use the `javax.persistence.criteria.*` imports instead of the `org.hibernate.criterion.*` ones) . We should use either JPQL queries, or JPA criteria API.